pretyping/typeclasses.mli

```diff
@@ -54,19 +53,25 @@ type typeclass = {
       no backtracking and sharing of resolution. *)
 }
 
-type instance
+type instance = {
+  is_class: GlobRef.t;
+  is_info: hint_info;
+  (* Sections where the instance should be redeclared,
+     None for discard, Some 0 for none. *)
+  is_global: int option;
+  is_impl: GlobRef.t;
+}
```

Comment:
Can you explain why this is needed? (the last two fields seem not to be tampered)

Reply:
San't some of this data be move to vernac as well? I mean leave here the fields that are needed here, and move upward the ones for libstack management (assuming I got it right)

Reply:
You are right that we should improve the separation of concerns even more than this PR does. I plan to do it as part of the libobject redesign. Roughly, instead of relying on inverted control spread everywhere in the code, we should compile vernac ASTs to vernac actions, in direct style. It will be easier to clarify the interfaces at that point.